### PR TITLE
nova/metrics: Fix empty instances gauge

### PR DESCRIPTION
**Pull request by grandchild**
The query used the disused and now mostly-empty `instance_types`
table, which made the metric report no values (or only a very few
ancient instances in some regions).

Use instance_extra instead and parse the flavor name out of the JSON.
The query is a bit more expensive this way (it doubled in time from
the query without any join).

### CODE REVIEW

**Comment by Carthaca** [comment]
Since you are reactivating a non-working metric, may I suggest to think about another change: exclude soft-deleted entries.

Otherwise our customers would always have to do the extra work of filtering to get the same instance list like they get via API. And I'm not aware of use-cases for having deleted instances.

File: openstack/nova/values.yaml [hunk 1]
```diff
@@ -530,11 +530,11 @@ mysql_metrics:
           instances.project_id,
           instances.hostname,
           instances.vm_state,
-          coalesce(instance_types.name,'N/A') AS flavor_name,
+          coalesce(JSON_VALUE(instance_extra.flavor, '$.cur."nova_object.data".name'),'N/A') AS flavor_name
```
Comment (Carthaca) [inline openstack/nova/values.yaml:533]: missing a comma at the end

**Review by Carthaca — CHANGES_REQUESTED** [review]

File: openstack/nova/values.yaml (hunk 1 above)
Comment (grandchild) [reply openstack/nova/values.yaml:533]: ooh, well spotted!

**Review by Carthaca — APPROVED** [review]

**Comment by grandchild** [comment]
sorry, I added the non-deleted instances clause in a separate commit. would you have a look again?

File: openstack/nova/values.yaml [hunk 2]
```diff
@@ -530,11 +531,12 @@ mysql_metrics:
...
           instances.hostname,
           instances.vm_state,
-          coalesce(instance_types.name,'N/A') AS flavor_name,
+          coalesce(JSON_VALUE(instance_extra.flavor, '$.cur."nova_object.data".name'),'N/A') AS flavor_name,
           COUNT(*) AS gauge
         FROM instances
-        JOIN instance_types ON instances.instance_type_id=instance_types.id where host LIKE 'nova-compute-ironic%'
-        GROUP BY instances.vm_state, instances.project_id, instance_types.name, instances.uuid, instances.hostname, instances.node;
+        LEFT JOIN instance_extra ON instances.uuid=instance_extra.instance_uuid WHERE host LIKE 'nova-compute-ironic%'
+        WHERE instances.deleted = 0
```
Comment (Carthaca) [inline openstack/nova/values.yaml:538]:
```suggestion
        AND instances.deleted = 0
```

Comment (Carthaca) [reply openstack/nova/values.yaml:538]: There is already a `WHERE` in the line above

**Review by Carthaca — CHANGES_REQUESTED** [review]

File: openstack/nova/values.yaml (hunk 2 above)
Comment (grandchild) [reply openstack/nova/values.yaml:538]: O.O thanks again!

**Review by Carthaca — APPROVED** [review]

**Pull request merged**